pkg/restic/common_test.go

```diff
@@ -0,0 +1,284 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restic
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+
+	corev1api "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arkv1api "github.com/heptio/ark/pkg/apis/ark/v1"
+	"github.com/heptio/ark/pkg/generated/clientset/versioned/fake"
+	informers "github.com/heptio/ark/pkg/generated/informers/externalversions"
+)
+
+func TestPodHasSnapshotAnnotation(t *testing.T) {
+	tests := []struct {
+		name        string
+		annotations map[string]string
+		expected    bool
+	}{
+		{
+			name:        "nil annotations",
+			annotations: nil,
+			expected:    false,
+		},
+		{
+			name:        "empty annotations",
+			annotations: make(map[string]string),
+			expected:    false,
+		},
+		{
+			name:        "non-empty map, no snapshot annotation",
+			annotations: map[string]string{"foo": "bar"},
+			expected:    false,
+		},
+		{
+			name:        "has snapshot annotation only, no suffix",
+			annotations: map[string]string{podAnnotationPrefix: "bar"},
+			expected:    true,
+		},
+		{
+			name:        "has snapshot annotation only, with suffix",
+			annotations: map[string]string{podAnnotationPrefix + "foo": "bar"},
+			expected:    true,
+		},
+		{
+			name:        "has snapshot annotation, with suffix",
+			annotations: map[string]string{"foo": "bar", podAnnotationPrefix + "foo": "bar"},
+			expected:    true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pod := &corev1api.Pod{}
+			pod.Annotations = test.annotations
+			assert.Equal(t, test.expected, PodHasSnapshotAnnotation(pod))
+		})
+	}
+}
+
+func TestGetPodSnapshotAnnotations(t *testing.T) {
+	tests := []struct {
+		name        string
+		annotations map[string]string
+		expected    map[string]string
+	}{
+		{
+			name:        "nil annotations",
+			annotations: nil,
+			expected:    nil,
+		},
+		{
+			name:        "empty annotations",
+			annotations: make(map[string]string),
+			expected:    nil,
+		},
+		{
+			name:        "non-empty map, no snapshot annotation",
+			annotations: map[string]string{"foo": "bar"},
+			expected:    nil,
+		},
+		{
+			name:        "has snapshot annotation only, no suffix",
+			annotations: map[string]string{podAnnotationPrefix: "bar"},
+			expected:    map[string]string{"": "bar"},
+		},
+		{
+			name:        "has snapshot annotation only, with suffix",
+			annotations: map[string]string{podAnnotationPrefix + "foo": "bar"},
+			expected:    map[string]string{"foo": "bar"},
+		},
+		{
+			name:        "has snapshot annotation, with suffix",
+			annotations: map[string]string{"foo": "bar", podAnnotationPrefix + "foo": "bar", podAnnotationPrefix + "abc": "123"},
```

Comment:
Let's use something besides foo/bar for the non-prefixed annotation, just to be safe.

Reply:
done.